### PR TITLE
remove codepaths for async activity writes

### PR DESCRIPTION
**Pull request by davidsbailey**
### Background

prep work for fixing [LP-60](https://codedotorg.atlassian.net/browse/LP-60). supercedes https://github.com/code-dot-org/code-dot-org/pull/29263 . see [discussion in slack](https://codedotorg.slack.com/archives/C03CK49G9/p1561061485406200) describing the new approach.

### Description

1. remove dead code related to async activity writes. this code is considered dead because we currently have activity writes completely disabled via gatekeeper, as can be seen here:

    Screen Shot 2019-06-21 at 3 12 42 PM
```
[production] dashboard > Gatekeeper.allows('activities', where: {script_name: 'coursea-2018'}, default: true)
=> false
```

2. https://github.com/code-dot-org/code-dot-org/pull/29294/commits/811cdaea09f838dec58c8c751f9e9d121b84336c fixes broken logic related to confusion between a `UserLevel` object and the `@new_level_completed` boolean . This will likely cause us to stop over-reporting `new_level_completed` to Google Analytics here: https://github.com/code-dot-org/code-dot-org/blob/44640488480ec19761be5b53d7ac68d42c578f89/apps/src/StudioApp.js#L1139-L1142 and may also fix a problem with `save_to_gallery_url` here: https://github.com/code-dot-org/code-dot-org/blob/47cd42ad5ec4f602855b8be017a212d40f114281/dashboard/app/controllers/application_controller.rb#L203-L206

### Future work

* consider adding `script_id` column to `activities` table
* start recording activities for multi assessment levels 
* rip out gallery code. https://studio.code.org/gallery has no additions within the past year. already tracked at https://codedotorg.atlassian.net/browse/LP-415

### CODE REVIEW

**Comment by islemaster** [comment]
This is awesome!  Walked through each commit and your approach looks sound to me.  I was surprised to learn we don't need to support the async writes pathway anymore since it seems like it's been useful in the past.  Do you know if this is because of the switch to Aurora?

**Comment by davidsbailey** [comment]
> I was surprised to learn we don't need to support the async writes pathway anymore since it seems like it's been useful in the past. Do you know if this is because of the switch to Aurora?

My understanding is that we do not need the async pathways because we have not used them for several years now. In 2017 we turned off activity writes completely before Hour of Code, then turned them back on again after. In 2018 we turned off activity writes before Hour of Code, then did not turn them back on because we were approaching an id overflow situation. @wjordan may be able to provide more details.

**Review by islemaster — APPROVED** [review]

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -1745,38 +1745,6 @@ def self.track_proficiency(user_id, script_id, level_id)
...
-      User.handle_async_op(async_op)
-    end
-
-    old_result = old_user_level.try(:best_result)
-    !ActivityConstants.passing?(old_result) && ActivityConstants.passing?(new_result)
-  end
-
   # The synchronous handler for the track_level_progress helper.
   # @return [UserLevel]
   def self.track_level_progress_sync(user_id:, level_id:, script_id:, new_result:, submitted:, level_source_id:, pairing_user_ids: nil, is_navigator: false)
```
Comment (wjordan) [inline dashboard/app/models/user.rb]: Could this be renamed/simplified to remove the `_sync` now that there is no `_async` counterpart?

**Review by wjordan — APPROVED** [review]
This looks excellent, I love how much it simplifies this section of the codebase with all of the removed code. This has been a cleanup task we've been looking forward to doing for years, thanks for getting it done!

**Comment by davidsbailey** [comment]
Excellent, thanks for steering us toward this direction, @wjordan ! I'm happy to have an excuse to remove some dead code.

**Pull request merged**